### PR TITLE
Added support for ThrowableProblem.getCause

### LINKED ISSUE #3: Support ThrowableProblem.getCause

### PR DESCRIPTION
**Pull request by whiskeysierra**
This closes #3

### CODE REVIEW

**Comment by lukasniemeier-zalando** [comment]
Merge and release?

**Comment by whiskeysierra** [comment]
> Merge and release?

What about a new unwritten law that requires at least one :+1: on a pull request before it get's merged?

File: problem/src/main/java/org/zalando/problem/ThrowableProblem.java [hunk 1]
```diff
@@ -22,9 +22,29 @@
...
+    }
+
+    @Override
+    public ThrowableProblem getCause() {
+        final Throwable cause = super.getCause();
+        if (cause == null) {
+            return null;
+        }
 
+        checkState(cause instanceof ThrowableProblem, "Expected throwable problem cause, but got %s", cause);
```
Comment (jhorstmann) [inline problem/src/main/java/org/zalando/problem/ThrowableProblem.java]: Can we check the cause in the constructor instead? That would give a better stacktrace to find the error.

Comment (whiskeysierra) [inline problem/src/main/java/org/zalando/problem/ThrowableProblem.java]: > Can we check the cause in the constructor instead? That would give a better stacktrace to find the error.

I changed the parameter type to `ThrowableProblem`

**Comment by whiskeysierra** [comment]
@jhorstmann I simplified the `getCause` method (including the cast) and added comment. The correct type is now ensured via the constructor parameter. There was no need for an explicit `instanceof` check anymore.

**Comment by whiskeysierra** [comment]
Any objections for merge+release?

File: jackson-datatype-problem/pom.xml [hunk 2]
```diff
@@ -75,6 +75,12 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.slf4j</groupId>
```
Comment (lukasniemeier-zalando) [inline jackson-datatype-problem/pom.xml]: :+1:

**Comment by lukasniemeier-zalando** [comment]
:+1: go!

**Pull request merged**